### PR TITLE
[SYCL][CUDA] Use __nvvm_reflect_ocl instead of LLVM IR workaround

### PR DESCRIPTION
**Pull request by hdelan**
We needed an IR workaround in order to use `__nvvm_reflect` in OpenCL. This is because passing a string in OpenCL to `__nvvm_reflect` requires the extern declaration of `__nvvm_reflect` like so:

`int __nvvm_reflect(__constant char *)` 

Since strings in OpenCL are in the `__constant` address space. This then clashes with use of `__nvvm_reflect` within CUDA's libdevice which uses the generic address space to pass in the string.

Therefore a new func name was needed to allow nvvm reflect capabilities, but while using a ptr input in different address space. This change was made upstream see https://github.com/intel/llvm/commit/ce43e2f07477a166712b7ee4a89fbc65d2ed56ad and has been available in DPC++ since the pulldown.

Therefore in OpenCL we can now use `__nvvm_reflect_ocl` without worrying about how it will clash with nvvm libdevice's use of `__nvvm_reflect`. Removing the need for the IR workaround in `reflect.ll`

I have also changed the `.gitignore` just since it is a small uncontroversial change so we can do in-tree tests in a local build dir of `sycl-e2e`

### CODE REVIEW

**Review by jchlanda — APPROVED** [review]

**Review by sergey-semenov — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,7 @@
 
 # Nested build directory
 /build*
+/sycl/test-e2e/build*
```
Comment (bader) [inline .gitignore:31]: This change is not related to `__nvvm_reflect_ocl` built-in. Please, move to a separate PR.

I don't understand why we need this change.

Comment (hdelan) [reply .gitignore]: `.gitignore` change removed

Comment (hdelan) [reply .gitignore]: This change does not change any behaviours of libclc but it is needed for "upstreamability", since it's better practice to have a libclc written in just openCL instead of needing to mix LLVM IR and openCL which compile into the same bc libs.

**Review by bader — APPROVED** [review]

**Comment by bader** [comment]
@hdelan, could you resolve merge conflicts, please?

**Comment by hdelan** [comment]
@bader merge conflicts resolved

**Comment by bader** [comment]
@aelovikov-intel, one check failed with the error message not related to the change. Is this safe to merge? I guess this failure doesn't block any other checks.
```
Cleaning D:\github\_work\llvm\llvm/*
rm: cannot remove 'D:\github\_work\llvm\llvm/build-e2e/Plugin': Device or resource busy
node:child_process:9[3](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:3)5
    throw err;
    ^

Error: Command failed: rm -rf D:\github\_work\llvm\llvm/*
rm: cannot remove 'D:\github\_work\llvm\llvm/build-e2e/Plugin': Device or resource busy

    at checkExecSyncError (node:child_process:861:11)
    at Object.execSync (node:child_process:932:15)
    at Object.<anonymous> (D:\github\_work\llvm\llvm\src\devops\actions\cleanup\cleanup.js:3:26)
    at Module._compile (node:internal/modules/cjs/loader:1198:1[4](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:4))
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:12[5](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:5)2:10)
    at Module.load (node:internal/modules/cjs/loader:107[6](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:6):32)
    at Function.Module._load (node:internal/modules/cjs/loader:911:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:22:4[7](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:7) {
  status: 1,
  signal: null,
  output: [
    null,
    Buffer(0) [Uint[8](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:8)Array] [],
    Buffer(88) [Uint8Array] [
      114, 10[9](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:9),  58,  32,  99,  97, 1[10](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:10), [11](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:11)0, 111, 116,  32, 114,
      101, 109, 111, 118, 101,  32,  39,  68,  58,  92, 103, 105,
      116, 104, 117,  98,  92,  95, 119, 111, 114, 107,  92, 108,
      108, 118, 109,  92, 108, 108, 118, 109,  47,  98, 117, 105,
      108, 100,  45, 101,  50, 101,  47,  80, 108, 117, 103, 105,
      110,  39,  58,  32,  68, 101, 118, 105,  99, 101,  32, 111,
      114,  32, 114, 101, 115, 111, 117, 114,  99, 101,  32,  98,
      117, 115, [12](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:12)1,  10
    ]
  ],
  pid: 7776,
  stdout: Buffer(0) [Uint8Array] [],
  stderr: Buffer(88) [Uint8Array] [
    1[14](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:14), 109,  58,  32,  99,  97, 110, 110, 111, 116,  32, 114,
    101, 109, 111, 118, 101,  32,  39,  68,  58,  92, 103, 105,
    116, 104, 117,  98,  92,  95, 119, 111, 114, 107,  92, 108,
    108, 118, 109,  92, 108, 108, 118, 109,  47,  98, 117, 105,
    108, 100,  45, 101,  50, 101,  47,  80, 108, 117, 103, 105,
    110,  39,  58,  32,  68, 101, 118, 105,  99, 101,  32, 111,
    114,  32, 114, 101, 1[15](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:15), 111, 1[17](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:17), 114,  99, 101,  32,  98,
    117, 115, 1[21](https://github.com/intel/llvm/actions/runs/6157142231/job/16707305657?pr=9250#step:31:21),  10
  ]
}
```
